service/matching/matcher_data.go

```diff
@@ -552,9 +556,9 @@ func (w *waitableMatchResult) initMatch(d *matcherData) {
 // call with matcherData.lock held.
 // w.matchResult must be nil (can't call wake twice).
 // w must not be in queues anymore.
-func (w *waitableMatchResult) wake(res *matchResult) {
-	bugIf(w.matchResult != nil, "bug: wake called twice")
-	bugIf(w.matchHeapIndex >= 0, "bug: wake called but still in heap")
+func (w *waitableMatchResult) wake(logger log.Logger, res *matchResult) {
+	softassert.That(logger, w.matchResult == nil, "wake called twice")
+	softassert.That(logger, w.matchHeapIndex < 0, "wake called but still in heap")
```

Comment:
Only ones left without a mitigation; not sure what that would be?

Reply:
I think it's fine to just do nothing and let it wake again